### PR TITLE
add support for nested column segmentation

### PR DESCRIPTION
**Pull request by notthattal**
## Description
- Removes the requirements for 'not is_nested_col' from _infer_schema
- Adds a static function to the AggregationFunction class which extracts the inner-most column from a nested column (e.g. top_level.nested_col would return nested_col) since duckdb flattens out nested columns to only include the innermost one
- Adds calls to this new function for both Numeric and Sketch Aggregations

## Jira Ticket
https://arthurai.atlassian.net/browse/UP-3075

## Partner MRs
TBD

### CODE REVIEW

File: src/arthur_common/aggregations/aggregator.py [hunk 1]
```diff
@@ -35,6 +35,18 @@ def reported_aggregations() -> list[BaseReportedAggregation]:
         """Returns the list of aggregations reported by the aggregate function."""
         raise NotImplementedError
 
+    @staticmethod
+    def get_innermost_segmentation_columns(segmentation_cols: list[str]) -> list[str]:
+        """
+        Extracts the innermost column name for nested segmentation columns or 
+        returns the top-level column name for non-nested segmentation columns.
+        """
+        for i, col in enumerate(segmentation_cols):
```
[the remark this answers is not in the document]
Comment (notthattal) [reply src/arthur_common/aggregations/aggregator.py:44]: updated [here](https://github.com/arthur-ai/arthur-common/pull/52/commits/372b93592305d3e5bcfa96e57f3b7231f260bec8) and updated the test case in [this commit](https://github.com/arthur-ai/arthur-engine/pull/525/commits/5da3e3cd5cdd8ebb4d8f198b48611143fca3e23f) in the arthur-engine to test for it

File: src/arthur_common/aggregations/aggregator.py [hunk 2]
```diff
@@ -38,13 +38,19 @@ def reported_aggregations() -> list[BaseReportedAggregation]:
...
         """
-        Extracts the innermost column name for nested segmentation columns or 
+        Extracts the innermost column name for nested segmentation columns or
         returns the top-level column name for non-nested segmentation columns.
         """
         for i, col in enumerate(segmentation_cols):
-            nested_cols = col.split('.')
-            segmentation_cols[i] = nested_cols[-1]
-        
+            nested_cols = col.split('"')
```
Comment (alexshanabrook) [inline src/arthur_common/aggregations/aggregator.py:45]: Thanks Tal! Only note about this now is that this will also miss some edge cases:

Double quotes are also valid in column names. If you look at the `escape_identifier` function, then you can see any double quotes in the column name are escaped with an additional double quote.

 Eg. to escape `my"column_name`, we would query that column via `"my""column_name"`
 
 In this current implementation, `nested_cols` would become `["my", "", "column_name"]`, in which case you would return `column_name` even though the original name wasn't referring to a nested struct field.
 
 If you want to take a look at `unescape_identifier`, that might be a good place to start to figure out how to extract this field? Let me know what you think.
 
 It could be useful to also add unit tests in arthur-common for this extraction function to test some of the edge cases like this

Comment (notthattal) [reply src/arthur_common/aggregations/aggregator.py]: Fixed function [here](https://github.com/arthur-ai/arthur-common/pull/52/commits/76353ca2d617fae56ecd632e5f8852c97875ec4e)

Added test cases [here](https://github.com/arthur-ai/arthur-common/pull/52/commits/ff26079f9b8c1849adac7066965cd133f5bdc11e)

File: src/arthur_common/aggregations/aggregator.py [hunk 3]
```diff
@@ -42,15 +42,20 @@ def get_innermost_segmentation_columns(segmentation_cols: list[str]) -> list[str
...
             # otherwise return the name since it's a top-level column
-            if len(nested_cols) >= 3:
-                segmentation_cols[i] = nested_cols[-2]
+            if col.startswith('"') and col.endswith('"'):
+                identifier = col[1:-1]
+                identifier_split_in_struct_fields = re.split(r'"\."', identifier)
+                
+                if len(identifier_split_in_struct_fields) > 1:
+                    innermost_field = identifier_split_in_struct_fields[-1]
+                    segmentation_cols[i] = innermost_field.replace('""', '"')
```
Comment (alexshanabrook) [inline src/arthur_common/aggregations/aggregator.py:54]: nit: can we add a comment explaining why we're doing this? I think this code will be confusing in the future otherwise

File: tests/unit/aggregation_functions/test_aggregation_function.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            ['"test.col_with"quotes"."properly"returns"innermost"column"'],
+            ['properly"returns"innermost"column'],
+        ),
+    ],
+)
+def test_get_innermost_segmentation_columns(
+    segmentation_cols: List[str],
+    expected_segmentation_columns: List[str],
+) -> None:
+    result = AggregationFunction.get_innermost_segmentation_columns(segmentation_cols)
```
Comment (alexshanabrook) [inline tests/unit/aggregation_functions/test_aggregation_function.py:45]: thanks for adding these! looks great to me

**Review by alexshanabrook — DISMISSED** [review]

**Review by alexshanabrook — APPROVED** [review]
